runtime/src/main/java/org/corfudb/runtime/LiteRoutingQueueListener.java

```diff
@@ -72,118 +74,129 @@ public LiteRoutingQueueListener(CorfuStore corfuStore, String sourceSiteId, Stri
 
     @Override
     public void onNext(CorfuStreamEntries results) {
-        log.debug("LRQListener received {} updates at address {}!",
-                results.getEntries().size(), results.getTimestamp());
-        List<CorfuStreamEntry> entries = results.getEntries().entrySet().stream()
-                .map(Map.Entry::getValue).findFirst().get();
+        log.debug("LRQListener received {} updates at address {}!", results.getEntries().size(), results.getTimestamp());
+        List<CorfuStreamEntry> entries = results.getEntries().entrySet().stream().map(Map.Entry::getValue)
+                .findFirst().get();
+
         List<Table.CorfuQueueRecord> allEntries = new ArrayList<>(entries.size());
-        Queue.ReplicationType currentType = Queue.ReplicationType.LAST_FULL_SYNC_ENTRY;
+        List<Queue.RoutingTableEntryMsg> allRQMsgs = new ArrayList<>();
+
         for (CorfuStreamEntry entry : entries) {
-            if (!entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+            // The Source always 'adds' replicated data to the queue.  So ignore all operations where type != UPDATE.
+            // Non-update entries are written by LR (CLEAR) and this listener itself (DELETE after successful
+            // processing).
+            if (entry.getOperation() != CorfuStreamEntry.OperationType.UPDATE) {
                 continue;
             }
             Queue.CorfuGuidMsg key = (Queue.CorfuGuidMsg) entry.getKey();
             Queue.RoutingTableEntryMsg msg = (Queue.RoutingTableEntryMsg) entry.getPayload();
+            allRQMsgs.add(msg);
+
             Queue.CorfuQueueMetadataMsg metadataMsg = (Queue.CorfuQueueMetadataMsg) entry.getMetadata();
             allEntries.add(new Table.CorfuQueueRecord(key, metadataMsg, msg));
-            if (currentType.equals(Queue.ReplicationType.LAST_FULL_SYNC_ENTRY)) {
-                currentType = msg.getReplicationType();
-            } else if (!currentType.equals(msg.getReplicationType())) {
```

Comment:
just wondering if this check would still make sense to have? should we throw some error if mixed event types are delivered outside of the LAST_SNAPSHOT_SYNC_ENTRY case?

Reply:
yes it will be valuable to have such a check.  All replication types are written in a separate transaction(including LAST_SNAPSHOT_SYNC_ENTRY).

Added a change to capture the first message in the batch and validate the replication type for all subsequent messages.